File: 0x08-python-more_classes/1-rectangle.py
#!/usr/bin/python3


"""Defining a class Rectangle"""


class Rectangle:
    """
    Class that is used to define properties of a
     rectangle.
    """

    def __init__(self, width=0, height=0):
        """Creates new instances of a Rectangle.
        """
        self.width = 0
        self.height = 0
        self.height = height
        self.width = width

    @property
    def width(self):
        """retries the width
        """
        return self.__width

    @property
    def height(self):
        """retrieves the height
        """
        return self.__height

    @width.setter
    def width(self, value):
        """Property setter
        """
        if not isinstance(value, int):
            raise TypeError("width must be an integer")
        if value < 0:
            raise ValueError("width must be >= 0")
        self.__width = value

    @height.setter
    def height(self, value):
        """Property setter
        """
        if not isinstance(value, int):
            raise TypeError("height must be an integer")
        if value < 0:
            raise ValueError("height must be >= 0")
        self.__height = value

File: 0x13-javascript_objects_scopes_closures/1-rectangle.js
#!/usr/bin/node

/*
 * a class Rectangle that defines a rectangle:
 * You must use the class notation for defining your class
 * The constructor must take 2 arguments w and h
*/

class Rectangle {
  constructor (w, h) {
    if (w === 0 || h === 0 || w < 0 || h < 0) {
      return {};
    } else {
      this.width = w;
      this.height = h;
    }
  }
}

module.exports = Rectangle;
